sklearn/utils/_array_api.py

```diff
@@ -104,6 +167,20 @@ def isdtype(self, dtype, kind):
         return isdtype(dtype, kind, xp=self._namespace)
 
 
+def _check_device_cpu(device):  # noqa
+    if device not in {"cpu", None}:
+        raise ValueError(f"Unsupported device for NumPy: {device!r}")
+
+
+def _accept_device_cpu(func):
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        _check_device_cpu(kwargs.pop("device", None))
+        return func(*args, **kwargs)
+
+    return wrapped_func
+
+
 class _NumPyAPIWrapper:
```

Comment:
nit: This is more a philosophical question/for my education: do we have to prefix things with a `_` in a module that has a `_` prefix to indicate that they are private? Somehow it feels like "stuff in a private module is of course private". (Mostly this just triggers my "too many underscores" spidery senses, not sure it makes much difference in practice). Same comment applies to other things that are `_` prefixed here.

Reply:
I prefer to include the `_` prefix even when the module is private. For me, the `_` prefix makes a easier to tell from the call-site that something is private.

Reply:
Ja I don't know. I guess this is one of these style/taste questions to which there is no correct answer (at least a "correct" answer we'd all agree to :D). My strongest argument for dropping the `_` is that it is ugly, not a very persuasive argument ;)

Marking this as resolved for now.